### DISCUSSION TITLE
How can I highlight hydrogen bond donors/acceptors?

### DISCUSSION THREAD

**Discussion by YoavShamir5** [rdkit/rdkit#7296] [Q&A]
Hello, I have two related questions:
1) Where can I find the source code for rdMolDescriptors? I want to know how CalcNumHBD works.
2) How can I find the atom indices of hydrogen bond donors/acceptors reported by CalcNumHBD/CalcNumHBA?
Thanks!

**Comment by bp-kelley** [comment]
The code uses smarts patterns to determine the hydrogen bond donors and
acceptors

https://github.com/rdkit/rdkit/blob/eebad845f551b6b15f0655589a5ab67e52f8f53f/Code/GraphMol/Descriptors/Lipinski.cpp#L197

```
HBA =
***@***.***,N,P,S])]),$([nH0,o,s;+0])]"
HBD = "[N&!H0&v3,N&!H0&+1&v4,O&H1&+0,S&H1&+0,n&H1&+0]"
```

Hope this helps.

Cheers,
 Brian

**Reply by YoavShamir5** [reply]
Thanks a lot for this info, Brian. Do you know where can I take a look at the source code of rdMolDescriptors?
